### DISCUSSION TITLE
Model (sometimes) doesn't update changes in viewer until user interaction occurs

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#3601] [Q&A] [answered]
### Description

I am making some kind of configurator where you can change parts of the model by turning certain parts visibility on and off. The issue I am facing is that sometimes when I change a part it doesn't get updated in the viewer until the user interacts with the viewer (rotating or zooming in or out). I call updateFraming() and updateShadow() on every change.  I will attach a video showing the issue I am facing and a test link.

#### Live Demo

[screen-capture.webm](https://user-images.githubusercontent.com/92090684/180718298-335089f8-c602-4af7-b0fd-d19d014bf73a.webm)

### Version

- model-viewer: v1.12.0 (i think? i load it from here https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js)

### Browser Affected

- [ X ] Chrome, version: 103.0.5060.134
- [ X ] Edge
- [ ] Firefox
- [ ] IE
- [ X ] Safari 15.5

### OS

- [ X ] Android
- [ X ] iOS 15.5
- [ ] Linux
- [ ] MacOS
- [ X ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment, marked as answer]
You're outside of our public API by accessing our internal $scene, so the usual caveats apply. However, I think what you need is to set `mv[$scene].queueRender();` when you change something. I'm looking at ways to add an actual public API to support these kinds of flows, so stay tuned.

**Reply by danilokrulj** [reply]
Looks like that did the trick! Thanks a lot, you are a real life saver!!!
